diff --git a/src/main/kotlin/org/springframework/data/core/PropertyReferenceExtensions.kt b/src/main/kotlin/org/springframework/data/core/PropertyReferenceExtensions.kt
index 0d2003ba..35b29388 100644
--- a/src/main/kotlin/org/springframework/data/core/PropertyReferenceExtensions.kt
+++ b/src/main/kotlin/org/springframework/data/core/PropertyReferenceExtensions.kt
@@ -30,8 +30,8 @@ import kotlin.reflect.jvm.javaGetter
  *
  * @since 4.1
  */
-fun <T : Any, P : Any, N : Any> PropertyReference<T, P>.then(next: KProperty1<T, P?>): TypedPropertyPath<T, N> {
-	val nextPath = KPropertyReference.of<T, P>(next) as PropertyReference<P, N>
+fun <T : Any, P : Any, N : Any> PropertyReference<T, out P?>.then(next: KProperty1<P, N?>): TypedPropertyPath<T, N> {
+	val nextPath = KPropertyReference.of<P, N>(next) as PropertyReference<P, N>
 	return TypedPropertyPaths.compose(this, nextPath)
 }
 
@@ -40,8 +40,8 @@ fun <T : Any, P : Any, N : Any> PropertyReference<T, P>.then(next: KProperty1<T,
  *
  * @since 4.1
  */
-fun <T : Any, P : Any, N : Any> PropertyReference<T, P>.then(next: KProperty<P?>): TypedPropertyPath<T, N> {
-	val nextPath = KPropertyReference.of<T, P>(next) as PropertyReference<P, N>
+fun <T : Any, P : Any, N : Any> PropertyReference<T, out P?>.then(next: KProperty<N?>): TypedPropertyPath<T, N> {
+	val nextPath = KPropertyReference.of<P, N>(next) as PropertyReference<P, N>
 	return TypedPropertyPaths.compose(this, nextPath)
 }
 
diff --git a/src/test/kotlin/org/springframework/data/core/PropertyReferenceKtUnitTests.kt b/src/test/kotlin/org/springframework/data/core/PropertyReferenceKtUnitTests.kt
index c12f79ea..1495d31d 100644
--- a/src/test/kotlin/org/springframework/data/core/PropertyReferenceKtUnitTests.kt
+++ b/src/test/kotlin/org/springframework/data/core/PropertyReferenceKtUnitTests.kt
@@ -37,10 +37,29 @@ class PropertyReferenceKtUnitTests {
 			.isThrownBy { PropertyReference.property(Person::address / Address::city) }
 	}
 
+	@Test
+	fun typedPropertyPathShouldCompose() {
+		val path = TypedPropertyPath.path(Person::address).then(Address::city)
+		assertThat(path.toDotPath()).isEqualTo("address.city")
+	}
+
+	@Test
+	fun kPropertyReferenceShouldCompose() {
+		val path = KPropertyReference.of(Person::address).then(Address::city)
+		assertThat(path.toDotPath()).isEqualTo("address.city")
+	}
+
+	@Test
+	fun collectionPropertyPathShouldCompose() {
+		val path = KPropertyReference.of(Person::addresses).then(Address::city)
+		assertThat(path.toDotPath()).isEqualTo("addresses.city")
+	}
+
 	class Person {
 		var name: String? = null
 		var age: Int = 0
 		var address: Address? = null
+		var addresses: List<Address> = emptyList()
 	}
 
 	class Address {
